### ISSUE TITLE
`no_effect` lints on usages of `trace_macros!`

### ISSUE THREAD

**Issue by cyrgani** [rust-lang/rust-clippy#15653]
### Summary

The `no_effect` lint fires on usages of the unstable `trace_macros!` macro within functions.
It seems as if the macro expands to `();` there.

### Lint Name

no_effect

### Reproducer

I tried this code:

```rust
#![feature(trace_macros)]

fn main() {
    trace_macros!(true);
}
```

I saw this happen:

```
warning: statement with no effect
 --> src/main.rs:4:5
  |
4 |     trace_macros!(true);
  |     ^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#no_effect
  = note: `#[warn(clippy::no_effect)]` on by default```

I expected to see this happen:
no output

### Version

```text
rustc 1.91.0-nightly (7ad23f43a 2025-09-09)
binary: rustc
commit-hash: 7ad23f43a225546c095123de52cc07d8719f8e2b
commit-date: 2025-09-09
host: x86_64-unknown-linux-gnu
release: 1.91.0-nightly
LLVM version: 21.1.0
```

### Additional Labels

@rustbot label P-low
(unstable feature with no path to stabilization)
